frontend/src/components/assessments/student-experience/QuestionNumbers.tsx

```diff
@@ -0,0 +1,50 @@
+import React, { useContext } from "react";
+import { GridItem, SimpleGrid } from "@chakra-ui/react";
+
+import StudentContext from "../../../contexts/StudentContext";
+import QuestionNumberTypes from "../../../types/QuestionNumberTypes";
+
+import QuestionNumber from "./QuestionNumber";
+
+const QuestionNumbers = (): React.ReactElement => {
+  const {
+    test,
+    answers,
+    currentQuestionIndex,
+    setCurrentQuestionIndex,
+  } = useContext(StudentContext);
+
+  const isCompletedQuestion = (index: number) => {
+    return answers[index].elements.every(
+      (element) => element.elementAnswers.length !== 0,
+    );
+  };
+
+  const questionStatus = (index: number) => {
```

Comment:
Would this ever be useful somewhere else? Feels like a `StudentUtils`-type thing but maybe not

Reply:
i don't think it will be used anywhere else (b/c this is the last ticket for the student stuff), but i agree utils is a good place to put it, so made the update!